src/api/delivery-needed/index.js

```diff
@@ -109,3 +127,128 @@ exports.deliveryNeededRequestHandler = async (req, res) => {
+  let error;
+
+  // validate request
+  try {
+    [request, error] = await findRequestByCode(requestCode);
+    if (error)
+      return res.status(404).send({
+        message: `${requestCode} not found.`,
+      });
+
+    if (request.fields.Status !== status_options.deliveryNeeded)
+      return res.status(400).send({
+        message: `Cannot claim delivery ${requestCode} with status "${request.fields.Status}".`,
+      });
+
+    if (
+      !request.fields[intakeVolunteer] ||
+      request.fields[intakeVolunteer].length === 0
+    )
+      return res.status(400).send({
+        message: `Cannot claim delivery ${requestCode} because there is no intake volunteer assigned. Please follow up with #intake_volunteers on Slack - we'll get it sorted out.`,
+      });
+  } catch (e) {
+    return res.status(400).send({ message: e });
+  }
+
+  // find volunteer records in volunteer table
+  try {
+    // delivery volunteer
+    [deliveryVol, error] = await findVolunteerByPhone(phoneNumber);
+
+    // TODO: on not found error, send volunteer form
+    if (!deliveryVol || (error && error.includes("404")))
+      return res.status(404).send({
+        message: error,
+      });
+    if (error) throw new Error(error);
+
+    // intake volunteer
+    [intakeVol, error] = await findVolunteerById(
+      request.fields[intakeVolunteer][0]
+    );
+
+    if (!intakeVol.fields[volunteersFields.phone])
+      error = `Cannot claim delivery ${requestCode} because the assigned intake volunteer has not provided their phone number. Please follow up with #intake_volunteers on Slack - we'll get it sorted out.`;
+
+    if (!intakeVol || (error && error.includes("404")))
+      return res.status(404).send({
+        message: error,
+      });
+    if (error) throw new Error(error);
+  } catch (e) {
+    return res.status(400).send({
+      message: `Something went wrong while looking for volunteer record: ${e}`,
+    });
+  }
+
+  try {
+    const {
+      [volunteersFields.phone]: deliveryPhone,
+      [volunteersFields.name]: deliveryName,
+    } = deliveryVol.fields;
+    const {
+      [volunteersFields.phone]: intakePhone,
+      [volunteersFields.phone]: intakeName,
+    } = intakeVol.fields;
+    await axios.post(TWILIO_SMS_DELIVERY_ENDPOINT, {
+      body: {
+        requestCode,
+        deliveryPhone,
+        deliveryName, // delivery volunteer name
+        intakePhone,
+        intakeName,
+      },
+    });
+  } catch (e) {
+    return res.status(400).send({
+      message: `Something went wrong while posting to Twilio: ${e}`,
+    });
+  }
+
+  // assign delivery volunteer to request
+  try {
+    // use email username in place of an empty volunteer name field
+    // return empty string if neither exist
+    const {
+      [volunteersFields.name]: volunteerName,
+      [volunteersFields.email]: volunteerEmail,
+    } = deliveryVol.fields;
+    const name =
+      volunteerName || (volunteerEmail && volunteerEmail.split("@")[0]) || "";
```

Comment:
in the body of the request that I tested (09/20) the deliveryName was not there, coming up "undefined" ...